### PR TITLE
actions: Guard GCC stuff

### PR DESCRIPTION
**Pull request by frostg-012**
## Type
- [x] bug fix 

- [ ] feature add 

- [ ] other 

## Checkbox
- [x] do not have break changes 

- [x] normal operation will not be affected after modification 

## Linked issues
<!-- if want to fix it, try "fixes: #13">

### CODE REVIEW

**Review by Fyg369 — COMMENTED** [review]
ok

**Comment by dabao1955** [comment]
> This code is to obtain the gcc compiler file prefix. In fact, it is almost the same whether you add a judgment or not, but adding a judgment is also excellent.

So？

**Comment by Fyg369** [comment]
> > This code is to obtain the gcc compiler file prefix. In fact, it is almost the same whether you add a judgment or not, but adding a judgment is also excellent.
> 
> So？

It would be better to add judgment

**Comment by frostg-012** [comment]
> This code is to obtain the gcc compiler file prefix. In fact, it is almost the same whether you add a judgment or not, but adding a judgment is also excellent.

yeah, but u can check my commit message. u will got it when u compiling using clang. so better to add a guard for that.

**Comment by dabao1955** [comment]
> > > This code is to obtain the gcc compiler file prefix. In fact, it is almost the same whether you add a judgment or not, but adding a judgment is also excellent.
> > 
> > So？
> 
> It would be better to add judgment

```bash
         if [ -d $HOME/gcc-64/bin/ ] || [ -d $HOME/gcc-32/bin/ ]; then
```

We have a one.

**Comment by Fyg369** [comment]
> > > > This code is to obtain the gcc compiler file prefix. In fact, it is almost the same whether you add a judgment or not, but adding a judgment is also excellent.
> > > 
> > > 
> > > So？
> > 
> > 
> > It would be better to add judgment
> 
> which line？

**Comment by Fyg369** [comment]
> > > > > This code is to obtain the gcc compiler file prefix. In fact, it is almost the same whether you add a judgment or not, but adding a judgment is also excellent.
> > > > 
> > > > 
> > > > So？
> > > 
> > > 
> > > It would be better to add judgment
> > 
> > 
> > which line？

File: action.yml [hunk 1]
```diff
@@ -282,21 +282,23 @@ runs:
...
          cd kernel/${{ inputs.kernel-dir }}
-         for GCC_DIR in "$HOME/gcc-64" "$HOME/gcc-32"; do
-             find "$GCC_DIR"/*/*/bin -type d -exec sh -c 'mv "$(dirname "{}")"/* "$GCC_DIR"/' _ {} \; -quit > /dev/null 2>&1 || true
-             for FILE in "$GCC_DIR/bin"/*; do
-                 FILE_NAME=$(basename "$FILE")
-                 MATCHED_DIR=""
-                 for FOLDER in $(find "$GCC_DIR" -mindepth 1 -maxdepth 1 -type d | sort -r); do
-                     FOLDER_NAME=$(basename "$FOLDER")
-                     [[ "$FILE_NAME" == "$FOLDER_NAME"* ]] && MATCHED_DIR="$FOLDER_NAME" && break
+         if [ -d $HOME/gcc-64/bin/ ] || [ -d $HOME/gcc-32/bin/ ]; then
```
Comment (dabao1955) [inline action.yml:285]: @Fyg369 Need to add additional tests? If so, what should be added a check for?

**Review by dabao1955 — APPROVED** [review]

**Comment by Fyg369** [comment]
Feedback on a KernelSU patching problem. The patch.sh patch code in the warehouse seems to be unable to successfully patch my namespace.c file. In my kernel source namespace.c file, there is only the static inline bool may_mandlock(void) function code. This is the same as the insertion snippet introduced in the KernelSU official website, but the code in the patch.sh patch in the warehouse is in sed -i '/int ksys_umount(char __user \*name, int flags)/i \Insert patch, please have time to modify it. Thank you

**Comment by Fyg369** [comment]
static inline bool may_mandlock(void)
{
#ifndef CONFIG_MANDATORY_FILE_LOCKING
    return false;
#endif
    return capable(CAP_SYS_ADMIN);
} 
 // Insert new code here

**Comment by dabao1955** [comment]
> Feedback on a KernelSU patching problem. The patch.sh patch code in the warehouse seems to be unable to successfully patch my namespace.c file. In my kernel source namespace.c file, there is only the static inline bool may_mandlock(void) function code. This is the same as the insertion snippet introduced in the KernelSU official website, but the code in the patch.sh patch in the warehouse is in sed -i '/int ksys_umount(char __user \*name, int flags)/i \Insert patch, please have time to modify it. Thank you
> 
> 
> 
> 
> 

？

**Comment by Fyg369** [comment]
> > Feedback on a KernelSU patching problem. The patch.sh patch code in the warehouse seems to be unable to successfully patch my namespace.c file. In my kernel source namespace.c file, there is only the static inline bool may_mandlock(void) function code. This is the same as the insertion snippet introduced in the KernelSU official website, but the code in the patch.sh patch in the warehouse is in sed -i '/int ksys_umount(char __user *name, int flags)/i \Insert patch, please have time to modify it. Thank you
> > 
> > 
> 
> ？

There is no int ksys_umount(char __user *name, int flags) code in my kernel namespace.c file and the patch cannot be successfully patched.

**Comment by Fyg369** [comment]
下面代码里这行sed -i '/int ksys_umount(char __user \*name, int flags)/i \是否需要修改一下？

   fs/namespace.c)
        sed -i '/int ksys_umount(char __user \*name, int flags)/i \
#ifdef CONFIG_KSU\
static int can_umount(const struct path *path, int flags)\
{\
    struct mount *mnt = real_mount(path->mnt);\
\
    if (flags & ~(MNT_FORCE | MNT_DETACH | MNT_EXPIRE | UMOUNT_NOFOLLOW))\
        return -EINVAL;\
    if (!may_mount())\
        return -EPERM;\
    if (path->dentry != path->mnt->mnt_root)\
        return -EINVAL;\
    if (!check_mnt(mnt))\
        return -EINVAL;\
    if (mnt->mnt.mnt_flags & MNT_LOCKED) /* Check optimistically */\
        return -EINVAL;\
    if (flags & MNT_FORCE && !capable(CAP_SYS_ADMIN))\
        return -EPERM;\
    return 0;\
}\
\
int path_umount(struct path *path, int flags)\
{\
    struct mount *mnt = real_mount(path->mnt);\
    int ret;\
\
    ret = can_umount(path, flags);\
    if (!ret)\
        ret = do_umount(mnt, flags);\
\
    /* we must not call path_put() as that would clear mnt_expiry_mark */\
    dput(path->dentry);\
    mntput_no_expire(mnt);\
    return ret;\
}\
#endif\
' fs/namespace.c
        ;;

**Comment by dabao1955** [comment]
>  下面代码里这行sed -i '/int ksys_umount(char __user \*name, int flags)/i \是否需要修改一下？
> 
>    fs/namespace.c)
>         sed -i '/int ksys_umount(char __user \*name, int flags)/i \
> #ifdef CONFIG_KSU\
> static int can_umount(const struct path *path, int flags)\
> {\
>     struct mount *mnt = real_mount(path->mnt);\
> \
>     if (flags & ~(MNT_FORCE | MNT_DETACH | MNT_EXPIRE | UMOUNT_NOFOLLOW))\
>         return -EINVAL;\
>     if (!may_mount())\
>         return -EPERM;\
>     if (path->dentry != path->mnt->mnt_root)\
>         return -EINVAL;\
>     if (!check_mnt(mnt))\
>         return -EINVAL;\
>     if (mnt->mnt.mnt_flags & MNT_LOCKED) /* Check optimistically */\
>         return -EINVAL;\
>     if (flags & MNT_FORCE && !capable(CAP_SYS_ADMIN))\
>         return -EPERM;\
>     return 0;\
> }\
> \
> int path_umount(struct path *path, int flags)\
> {\
>     struct mount *mnt = real_mount(path->mnt);\
>     int ret;\
> \
>     ret = can_umount(path, flags);\
>     if (!ret)\
>         ret = do_umount(mnt, flags);\
> \
>     /* we must not call path_put() as that would clear mnt_expiry_mark */\
>     dput(path->dentry);\
>     mntput_no_expire(mnt);\
>     return ret;\
> }\
> #endif\
> ' fs/namespace.c
>         ;;

你可以开一个pr

**Pull request merged**